File: backend/app/routes/product.py
from fastapi import APIRouter, HTTPException, status
from app.models import (
    CreateProductRequest,
    CreateProductResponse
)
from app.services.firebase import firebase_service

router = APIRouter(
    prefix="/product",
    tags=["product"]
)


@router.post("/create", response_model=CreateProductResponse, status_code=status.HTTP_201_CREATED)
async def create_product(product_request: CreateProductRequest):
    """
    Create a new product

    This endpoint creates a new product in the products collection.
    The server generates a sequential product_id (prod_001, prod_002, ...).

    Args:
        product_request: Product creation data
            - name: Product name
            - price: Product price
            - description: Product description (optional)
            - image_url: Product image URL (optional)
            - tags: Product tags (optional)

    Returns:
        CreateProductResponse with pid (product_id)
    """
    try:
        product_data = product_request.model_dump()

        product_id = await firebase_service.create_product(product_data)

        return CreateProductResponse(pid=product_id)
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Error creating product: {str(e)}"
        )

File: frontend-kiosk/src/services/api/product.js
/**
 * @fileoverview Product API
 *
 * 제품 관련 API 호출 함수를 제공합니다.
 */

import { request } from "./client.js";

/**
 * 전체 제품 목록 조회
 *
 * @async
 * @returns {Promise<Array<Product>>} 제품 목록 [{ pid, name, price, description, image_url, tags }]
 *
 * @example
 * const products = await getProducts();
 * products.forEach(product => {
 *   console.log(product.pid, product.name, product.price);
 * });
 */
export async function getProducts() {
  return request("/products/");
}

/**
 * 특정 제품 상세 정보 조회
 *
 * @async
 * @param {string} productId - 제품 ID (pid)
 * @returns {Promise<Product>} 제품 상세 정보 { pid, name, price, description, image_url, tags }
 *
 * @example
 * const product = await getProductById('prod_001');
 * console.log(product.name, product.price);
 */
export async function getProductById(productId) {
  return request(`/products/${productId}`);
}
